core/src/main/java/org/elasticsearch/bootstrap/JarHell.java

```diff
@@ -153,23 +149,34 @@ static void checkManifest(Manifest manifest, Path jar) {
         }
     }
 
+    public static void checkVersionFormat(String targetVersion) {
+        if (!targetVersion.matches("^0*[0-9]+(\\.[0-9]+)*$")) {
+            throw new IllegalStateException(
+                    String.format(
+                            Locale.ROOT,
+                            "version string must be a sequence of nonnegative decimal integers separated by \".\"'s and may have leading zeros but was %s",
+                            targetVersion
+                    )
+            );
+        }
+    }
+
     /**
      * Checks that the java specification version {@code targetVersion}
      * required by {@code resource} is compatible with the current installation.
      */
     public static void checkJavaVersion(String resource, String targetVersion) {
```

Comment:
It seems to me that changing `checkVersionFormat` to `checkVersionCompatible` would present some confusion between `checkVersionCompatible` and `checkJavaVersion`. The former is checking the format of the version, and the latter is checking if the target version is compatible with the current runtime. I'm certainly open to better names though.

However, I do like keeping these methods separate however as then we can have tests for the different functionality (checking the version format and checking version compatibility).